runtime/src/main/java/com/flipkart/flux/guice/module/AuthNModule.java

```diff
@@ -0,0 +1,107 @@
+                              @Named("uiAuthConfig.clientSecret") String clientSecret,
+                              @Named("uiAuthConfig.redirectUri") String redirectUrl,
+                              @Named("uiAuthConfig.authIgnoreUrls") String authIgnoreUrls
+    ) {
+        String[] authnUrls = new String[1];
+        authnUrls[0] = url;
+        AuthConfig authConfig = new AuthConfig();
+        authConfig.setAuthEnabled(authEnabled);
+        authConfig.setAuthnUrls(authnUrls);
+        authConfig.setClientId(clientId);
+        authConfig.setClientSecret(clientSecret);
+        authConfig.setRedirectUri(redirectUrl);
+        authConfig.setAuthIgnoreUrls(authIgnoreUrls);
+        return authConfig;
+    }
+
+    @Provides
+    @Named("ApiAuthnConfig")
+    AuthConfig getAuthNConfig(@Named("apiAuthConfig.authEnabled") boolean authEnabled,
+                              @Named("apiAuthConfig.authnUrl") String url,
+                              @Named("apiAuthConfig.clientId") String clientId,
+                              @Named("apiAuthConfig.clientSecret") String clientSecret,
+                              @Named("apiAuthConfig.authIgnoreUrls") String authIgnoreUrls
+
+    ) {
+        String[] authnUrls = new String[1];
+        authnUrls[0] = url;
+        AuthConfig authConfig = new AuthConfig();
+        authConfig.setAuthEnabled(authEnabled);
+        authConfig.setAuthnUrls(authnUrls);
+        authConfig.setClientId(clientId);
+        authConfig.setClientSecret(clientSecret);
+        authConfig.setAuthIgnoreUrls(authIgnoreUrls);
+        return authConfig;
+    }
+
+
+    public static void configureUIApp(WebAppContext webAppContext, AuthConfig config) {
+        if (!config.isAuthEnabled()) {
+            return;
+        }
+
+        String clientId = config.getClientId();
+        String clientSecret = config.getClientSecret();
+        LoginUrlFilter loginUrlFilter = new LoginUrlFilter(config.getAuthnUrls()[0], clientId, config.getScopes(), config.getAdditionalParams());
+
+        CallbackUrlFilter callbackUrlFilter = new CallbackUrlFilter(config.getAuthnUrls()[0], clientId, clientSecret);
+        LogoutUrlFilter logoutUrlFilter = new LogoutUrlFilter(config.getAuthnUrls()[0]);
+
+        SecurityContextPersistenceFilter securityContextFilter = new SecurityContextPersistenceFilter();
+
+        loginUrlFilter.setLoginUrl(config.getLoginUrl());
+        callbackUrlFilter.setRedirectUrl(config.getLoginUrl() + "_callback");
+
+        if (config.getRedirectUri() != null) {
+            loginUrlFilter.setRedirectUri(config.getRedirectUri());
+            callbackUrlFilter.setRedirectUri(config.getRedirectUri());
+        }
+
+        webAppContext.addFilter(new FilterHolder(securityContextFilter), "/*", EnumSet.of(DispatcherType.REQUEST));
+        webAppContext.addFilter(new FilterHolder(loginUrlFilter), "/*", EnumSet.of(DispatcherType.REQUEST));
```

Comment:
Why Login and Logouts are implemented as filters? Shouldn't they be services? SecurityContextFilter can redirect to these service endpoints.

Reply:
Ohk! Looks like this is Authn Library code. got that.